engine/access/ping/engine.go

```diff
@@ -44,6 +47,20 @@ func New(
 		middleware:   mw,
 	}
 
+	// if a node info file is provided, it is read and the additional node information is reported as part of the ping metric
+	if nodeInfoFile != "" {
+		nodeInfo, err := readJson(nodeInfoFile)
+		if err != nil {
+			log.Error().Err(err).Str("node_info_file", nodeInfoFile).Msg("failed to read node info file")
+		} else {
+			eng.nodeInfo = nodeInfo
+			log.Debug().Str("node_info_file", nodeInfoFile).Msg("using node info file")
+		}
+	} else {
```

Comment:
I noticed we don't initialize `nodeInfo` as an empty map. Will this cause any nil pointer issues for cases when we don't pass in a file?

Reply:
I did test that earlier using a simple program - https://play.golang.org/p/pVlcnI0S7yV
But maybe for clarity I will just initialize it to an empty map.